### DISCUSSION TITLE
ROCm 6.0.0 Release

### DISCUSSION THREAD

**Discussion by saadrahim** [ROCm/ROCm#2729] [Announcements]
# Release notes for AMD ROCm™ 6.0

ROCm 6.0 is a major release with new performance optimizations, expanded frameworks and library
support, and improved developer experience. This includes initial enablement of the AMD Instinct™
MI300 series. Future releases will further enable and optimize this new platform. Key features include:

* Improved performance in areas like lower precision math and attention layers.
* New hipSPARSELt library accelerates AI workloads via AMD's sparse matrix core technique.
* Upstream support is now available for popular AI frameworks like TensorFlow, JAX, and PyTorch.
* New support for libraries, such as DeepSpeed, ONNX-RT, and CuPy.
* Prepackaged HPC and AI containers on AMD Infinity Hub, with improved documentation and
  tutorials on the [AMD ROCm Docs](https://rocm.docs.amd.com) site.
* Consolidated developer resources and training on the new
  [AMD ROCm Developer Hub](https://www.amd.com/en/developer/resources/rocm-hub.html).

The following section provide a release overview for ROCm 6.0. For additional details, you can refer to
the [Changelog](https://rocm.docs.amd.com/en/develop/about/CHANGELOG.html). We list known
issues on [GitHub](https://github.com/ROCm/ROCm/issues).

## OS and GPU support changes

ROCm 6.0 enables the use of MI300A and MI300X Accelerators with a limited operating systems
support. Future releases will add additional OS's to match our general offering.

| Operating Systems | MI300A | MI300X |
|:---:|:---:|:---:|
| Ubuntu 22.04.5 | Supported | Supported |
| RHEL 8.9 | Supported |  |
| SLES15 SP5 | Supported |  |

For older generations of supported Instinct products we've added the following operating systems:

* RHEL 9.3
* RHEL 8.9

Note: For ROCm 6.2 and beyond, we've planned for end-of-support (EoS) for the following operating
systems:

* Ubuntu 20.04.5
* SLES 15 SP4
* RHEL/CentOS 7.9

## New ROCm meta package

We've added a new ROCm meta package for easy installation of all ROCm core packages, tools, and
libraries. For example, the following command will install the full ROCm package: `apt-get install rocm`
(Ubuntu), or `yum install rocm` (RHEL).

## Filesystem Hierarchy Standard

ROCm 6.0 fully adopts the Filesystem Hierarchy Standard (FHS) reorganization goals. We've removed
the backward compatibility support for old file locations.

## Compiler location change

* The installation path of LLVM has been changed from `/opt/rocm-<rel>/llvm` to
  `/opt/rocm-<rel>/lib/llvm`. For backward compatibility, a symbolic link is provided to the old
  location and will be removed in a future release.
* The installation path of the device library bitcode has changed from `/opt/rocm-<rel>/amdgcn` to
  `/opt/rocm-<rel>/lib/llvm/lib/clang/<ver>/lib/amdgcn`. For backward compatibility, a symbolic link
  is provided and will be removed in a future release.

## Documentation

CMake support has been added for documentation in the
[ROCm repository](https://github.com/RadeonOpenCompute/ROCm).

## AMD Instinct™ MI50 end-of-support notice

AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively gfx906 GPUs) enters
maintenance mode in ROCm 6.0.

As outlined in [5.6.0](https://rocm.docs.amd.com/en/docs-5.6.0/release.html), ROCm 5.7 was the
final release for gfx906 GPUs in a fully supported state.

  * Henceforth, no new features and performance optimizations will be supported for the gfx906 GPUs.
  * Bug fixes and critical security patches will continue to be supported for the gfx906 GPUs until Q2
    2024 (end of maintenance \[EOM] will be aligned with the closest ROCm release).
  * Bug fixes will be made up to the next ROCm point release.
  * Bug fixes will not be backported to older ROCm releases for gfx906.
  * Distribution and operating system updates will continue per the ROCm release cadence for gfx906
    GPUs until EOM.

## ROCm projects

The following sections contains project-specific release notes for ROCm 6.0. For additional details, you
can refer to the [Changelog](https://rocm.docs.amd.com/en/develop/about/CHANGELOG.html).

### AMD SMI

* **Integrated the E-SMI (EPYC-SMI) library**.
    You can now query CPU-related information directly through AMD SMI. Metrics include power,
    energy, performance, and other system details.

* **Added support for gfx942 metrics**.
    You can now query MI300 device metrics to get real-time information. Metrics include power,
    temperature, energy, and performance.

### HIP

* **New features to improve resource interoperability**.
  * For external resource interoperability, we've added new structs and enums.
  * We've added new members to HIP struct `hipDeviceProp_t` for surfaces, textures, and device
    identifiers.

* **Changes impacting backward compatibility**.
    There are several changes impacting backward compatibility: we changed some struct members and
    some enum values, and removed some deprecated flags. For additional information, please refer to
    the Changelog.

### hipCUB

* **Additional CUB API support**.
    The hipCUB backend is updated to CUB and Thrust 2.1.

### HIPIFY

* **Enhanced CUDA2HIP document generation**.
    API versions are now listed in the CUDA2HIP documentation. To see if the application binary
    interface (ABI) has changed, refer to the
    [*C* column](https://rocm.docs.amd.com/projects/HIPIFY/en/latest/tables/CUDA_Runtime_API_functions_supported_by_HIP.html)
    in our API documentation.

* **Hipified rocSPARSE**.
    We've implemented support for the direct hipification of additional cuSPARSE APIs into rocSPARSE
    APIs under the `--roc` option. This covers a major milestone in the roadmap towards complete
    cuSPARSE-to-rocSPARSE hipification.

### hipRAND

* **Official release**.
    hipRAND is now a *standalone project*--it's no longer available as a submodule for rocRAND.

### hipTensor

* **Added architecture support**.
    We've added contraction support for gfx942 architectures, and f32 and f64 data
    types.

* **Upgraded testing infrastructure**.
    hipTensor will now support dynamic parameter configuration with input YAML config.

### MIGraphX

* **Added TorchMIGraphX**.
    We introduced a Dynamo backend for Torch, which allows PyTorch to use MIGraphX directly
    without first requiring a model to be converted to the ONNX model format. With a single line of
    code, PyTorch users can utilize the performance and quantization benefits provided by MIGraphX.

* **Boosted overall performance with rocMLIR**.
    We've integrated the rocMLIR library for ROCm-supported RDNA and CDNA GPUs. This
    technology provides MLIR-based convolution and GEMM kernel generation.

* **Added INT8 support across the MIGraphX portfolio**.
    We now support the INT8 data type. MIGraphX can perform the quantization or ingest
    prequantized models. INT8 support extends to the MIGraphX execution provider for ONNX Runtime.

### ROCgdb

* **Added support for additional GPU architectures**.
  * Navi 3 series: gfx1100, gfx1101, and gfx1102.
  * MI300 series: gfx942.

### rocm-smi-lib

* **Improved accessibility to GPU partition nodes**.
    You can now view, set, and reset the compute and memory partitions. You'll also get notifications of
    a GPU busy state, which helps you avoid partition set or reset failure.

* **Upgraded GPU metrics version 1.4**.
    The upgraded GPU metrics binary has an improved metric version format with a content version
    appended to it. You can read each metric within the binary without the full `rsmi_gpu_metric_t` data
    structure.

* **Updated GPU index sorting**.
    We made GPU index sorting consistent with other ROCm software tools by optimizing it to use
    `Bus:Device.Function` (BDF) instead of the card number.

### ROCm Compiler

* **Added kernel argument optimization on gfx942**.
    With the new feature, you can preload kernel arguments into Scalar General-Purpose Registers
    (SGPRs) rather than pass them in memory. This feature is enabled with a compiler option, which also
    controls the number of arguments to pass in SGPRs. For more information, see:
    [https://llvm.org/docs/AMDGPUUsage.html#preloaded-kernel-arguments](https://llvm.org/docs/AMDGPUUsage.html#preloaded-kernel-arguments)

* **Improved register allocation at -O0**.
    We've improved the register allocator used at -O0 to avoid compiler crashes (when the signature is
    'ran out of registers during register allocation').

* **Improved generation of debug information**.
    We've improved compile time when generating debug information for certain corner cases. We've
    also improved the compiler to eliminate compiler crashes when generating debug information.

### ROCmValidationSuite

* **Added GPU and operating system support**.
    We added support for MI300X GPU in GPU Stress Test (GST).

### Roc Profiler

* **Added option to specify desired Roc Profiler version**.
    You can now use rocProfV1 or rocProfV2 by specifying your desired version, as the legacy rocProf
    (`rocprofv1`) provides the option to use the latest version (`rocprofv2`).

* **Automated the ISA dumping process by Advance Thread Tracer**.
    Advance Thread Tracer (ATT) no longer depends on user-supplied Instruction Set Architecture (ISA)
    and compilation process (using ``hipcc --save-temps``) to dump ISA from the running kernels.

* **Added ATT support for parallel kernels**.
    The automatic ISA dumping process also helps ATT successfully parse multiple kernels running in
    parallel, and provide cycle-accurate occupancy information for multiple kernels at the same time.

### ROCr

* **Support for SDMA link aggregation**.
    If multiple XGMI links are available when making SDMA copies between GPUs, the copy is
    distributed over multiple links to increase peak bandwidth.

### rocThrust

* **Added Thrust 2.1 API support**.
    rocThrust backend is updated to Thrust and CUB 2.1.

### rocWMMA

* **Added new architecture support**.
    We added support for gfx942 architectures.

* **Added data type support**.
    We added support for f8, bf8, xf32 data types on supporting architectures, and for bf16 in the HIP RTC
    environment.

* **Added support for the PyTorch kernel plugin**.
    We added awareness of `__HIP_NO_HALF_CONVERSIONS__` to support PyTorch users.

### TransferBench (beta)

* **Improved ordering control**.
    You can now set the thread block size (`BLOCK_SIZE`) and the thread block order (`BLOCK_ORDER`)
    in which thread blocks from different transfers are run when using a single stream.

* **Added comprehensive reports**.
    We modified individual transfers to report X Compute Clusters (XCC) ID when `SHOW_ITERATIONS`
    is set to 1.

* **Improved accuracy in result validation**.
    You can now validate results for each iteration instead of just once for all iterations.

<hr /><em>This discussion was created from the release <a href='https://github.com/ROCm/ROCm/releases/tag/rocm-6.0.0'>ROCm 6.0.0 Release</a>.</em>

**Comment by Danny3** [comment]
It would be great if Debian was supported too!

**Reply by xiaobo1025** [reply]
你好，我最近在我的windaos10系统上安装ROCm一直没有成功，你可以给我说一下怎么安装吗？真的非常谢谢你。

**Comment by baryluk** [comment]


**Reply by xiaobo1025** [reply]
Hello, I have recently failed to install ROCm on my windaos10 system. Can you tell me how to install it? 
Thank you very much.

**Reply by xiaobo1025** [reply]
thank you very much,Can I install and use ROCm normally by installing a linux virtual machine on windows10?

**Reply by MKKnorr** [reply]
Hello @baryluk, sorry for the inconvenience. Could you point out which link exactly does not work?

The [docker images in the rocm ecosystem](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/how-to/docker.html#docker-images-in-the-rocm-ecosystem) section has links to [Docker Hub](https://hub.docker.com/u/rocm) and [Infinity Hub](https://www.amd.com/en/technologies/infinity-hub) which provide various images. The `rocm-terminal` image contains the bare minimum to build HIP applications.

**Reply by MKKnorr** [reply]
@xiaobo1025 please refer to the documentation at rocm.docs.amd.com which contains [instructions for installing the HIP SDK on Windows](https://rocm.docs.amd.com/projects/install-on-windows/en/latest/how-to/install.html)

**Reply by xiaobo1025** [reply]
@MKKnorr Thank you very much.Does this mean that I can deploy ROCm and run it on my windows 10?

**Reply by johnnynunez** [reply]
All rocm libraries still not be in windows.

**Reply by MKKnorr** [reply]
@xiaobo1025 To verify whether your specific Windows system is supported please refer to [System requirements (Windows)](https://rocm.docs.amd.com/projects/install-on-windows/en/latest/reference/system-requirements.html) when following the [installation guide](https://rocm.docs.amd.com/projects/install-on-windows/en/latest/how-to/install.html#hip-install-full). As already mentioned, there are some [limitations](https://rocm.docs.amd.com/projects/install-on-windows/en/latest/reference/system-requirements.html#component-support), as some math libraries are Linux exclusive

**Reply by johnnynunez** [reply]
there is a ticked opened because the link is wrong: https://www.amd.com/en/developer/rocm-hub/hip-sdk.html

**Reply by MKKnorr** [reply]
Thanks, I've just taken a look at the issue, however it doesn't specify where that link came from. This post above has the correct link to the [AMD ROCm Developer Hub](https://www.amd.com/en/developer/resources/rocm-hub.html), maybe that can help you. Otherwise https://rocm.docs.amd.com/ has many resources on how to install and use ROCm

**Reply by hiepxanh** [reply]
https://rocm.docs.amd.com/projects/install-on-windows/en/latest/how-to/install.html#hip-install-full
At this page, the link is 404 not found 
![image](https://github.com/ROCm/ROCm/assets/16970990/a173f099-8de5-4ab3-9019-ef101d686287)

**Reply by Sabrewarrior** [reply]
@MKKnorr None of the resources work to install ROCm on Windows.

All of these have dead links to either https://www.amd.com/content/amd/live-site/en/developer/resources/rocm-hub/hip-sdk or https://www.amd.com/en/developer/resources/rocm-hub/hip-sdk.html
[AMD ROCm™ Developer Hub](https://www.amd.com/en/developer/resources/rocm-hub.html)
[Install HIP SDK](https://rocm.docs.amd.com/projects/install-on-windows/en/latest/how-to/install.html)
[Windows quick-start installation guide](https://rocm.docs.amd.com/projects/install-on-windows/en/latest/index.html#hip-install-quick)

**Reply by MKKnorr** [reply]
Sorry for the late reply. Thank you for the information, can you check these links again? They should be working now.

**Reply by hiepxanh** [reply]
@MKKnorr  Aw, so late, I thought that AMD leave user behind aready. Anyway the link was work, thank you, but the result still 5.7.1. I expect it should be align with Unbutu linux doesn't it?

**Reply by Sabrewarrior** [reply]
Not all ROCm releases have windows binaries, so latest is currently 5.7.1. Wish they had build from source instructions for windows though.

**Reply by johnnynunez** [reply]
There are still some bugs to compile rocm6 with windows if you see the pull requests from AMDMIGraphX and MIOpen
https://github.com/ROCm/MIOpen/pull/2719
https://github.com/ROCm/AMDMIGraphX/pull/2469

**Reply by MKKnorr** [reply]
The windows release cycle differs from the linux release cycle. For more information, see https://rocm.docs.amd.com/projects/install-on-windows/en/develop/conceptual/release-versioning.html The list is not up to date with the linux releases, but currently 5.7 is the latest windows release, so rocm 6.0 is not officially supported on windows. @Sabrewarrior it is sadly not possible to build rocm for windows from source, as it is not completely open source

**Reply by ColorfulRhino** [reply]
@MKKnorr Thanks for this info! I had to dig pretty deep to find your comment that a) Windows versions are behind Linux versions and that b) building for Windows is not supported. I believe it would be a good idea to have this information somwhere more accessible as well as in the build documents.

**Reply by MKKnorr** [reply]
@ColorfulShire  I agree, that this information is not very visible. As for adding a note to the build documents, I can't really do much about that, but feel free to open an issue at https://github.com/ROCm/HIP or even a PR to its documentation

**Reply by YellowRoseCx** [reply]
Is this still true? On the page https://rocm.docs.amd.com/projects/install-on-windows/en/develop/conceptual/release-versioning.html#windows-builds-from-source , it says
> Windows builds from source
> Not all source code required to build Windows from source is available under a permissive open source license. We only provide Windows build instructions for projects that can be built from source on Windows using a toolchain that has closed source build prerequisites. The ROCm manifest file is not valid for Windows. AMD does not release a manifest or tag our components in Windows. **You can use corresponding Linux tags to build on Windows.**

It seems to elude that building HIP SDK/ROCm is possible for Windows, but that the information is not distributed by AMD for legal purposes.

**Comment by baryluk** [comment]
This page mention docker images, but then the link goes to generic ROCm docs starting page. If you follow the docker link there (wrongly named [Install Docker containers](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/how-to/docker.html) , as nobody uses term Install for docker containers), it does not list any docker images, or show how to run any of these docker containers. Even if you go to the repo, it does not list any images (other than generic rocm-terminal).

Frustrating.

**Comment by johnnynunez** [comment]
I was able to compile tensorflow and pytorch with rocm6 for the 7900xt. Good job!

**Reply by xiaobo1025** [reply]
@johnnynunez You are very good!Could you give me some help my friends?Thank you very very much!

**Reply by johnnynunez** [reply]
https://github.com/johnnynunez/rocm_lab

**Comment by orlovdumitru** [comment]
can someone help me why I get following error: 
> I tensorflow/core/common_runtime/gpu/gpu_device.cc:2266] Ignoring visible gpu device (device: 0, name: Radeon RX 7900 XT, pci bus id: 0000:08:00.0) with AMDGPU version : gfx1100. The supported AMDGPU versions are gfx1030gfx1100, gfx900, gfx906, gfx908, gfx90a, gfx940, gfx941, gfx942.


**My system**:
``` 
    Ubuntu 22.04
    kernel 6.5.0
    GPU: AMD Radeon RX 7900XT
    installed ROCm 6.0
```
I also see someone else have same issue: https://community.amd.com/t5/discussions/rocm-6-0-0-tensorflow-not-working-on-rx-7900-xt/td-p/656431

**Reply by markscsmith** [reply]
That weird gfx1030gfx1100 was the clue that found me the Tensorflow bug: https://github.com/ROCm/tensorflow-upstream/issues/2410

**Comment by briansp2020** [comment]
@orlovdumitru 
See [this script](https://gist.github.com/briansp2020/1e8c3e5735087398ebfd9514f26a0007) to get an idea of how to compile TF to work on 7900XT/XTX. I'm not sure why they don't fix this simple error properly. But it's not too difficult to build a whl file that will work with your graphics card.
